analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+  purrr::set_names(library_ids)
+
+# define classification files
+singler_results_dir <- file.path(module_base, "results", "aucell_singler_annotation")
+classification_file_names <- glue::glue("{library_ids}_singler-classifications.tsv")
+classification_files <- file.path(singler_results_dir, sample_ids, classification_file_names) |>
+  purrr::set_names(library_ids)
+
+# define output files
+cluster_results_dir <- file.path(module_base, "results", "clustering", sample_ids)
+cluster_results_dir |>
+  purrr::walk(fs::dir_create)
+
+cluster_file_names <- glue::glue("{library_ids}_cluster-results.tsv")
+cluster_output_files <- file.path(cluster_results_dir, cluster_file_names) |>
+  purrr::set_names(library_ids)
+```
+
+```{r}
+# read in both SCEs
+sce_list <- sce_files |>
+  purrr::map(readr::read_rds)
+
+# read in classification data frame with SingleR results and combine into one
+classification_df <- classification_files |>
+  purrr::map(readr::read_tsv) |>
+  dplyr::bind_rows(.id = "library_id")
+```
+
+```{r}
+# read in marker genes table
+marker_genes_df <- readr::read_tsv(marker_genes_file, show_col_types = FALSE) |>
+  # account for genes being from multiple sources
+  dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |>
+  dplyr::distinct()
+
+# get list of all cell types found
+cell_types <- unique(marker_genes_df$cell_type)
+
+# get the sum of expression of all genes for each cell type
+gene_exp_df <- sce_list |>
+  purrr::map(\(sce) {
+    cell_types |>
+      purrr::map(\(type){
+        calculate_sum_markers(marker_genes_df, sce, type)
+      }) |>
+      purrr::reduce(dplyr::inner_join, by = "barcodes")
+  }) |>
+  dplyr::bind_rows(.id = "library_id")
+
+# join sum expression columns with classification df
+classification_df <- classification_df |>
+  dplyr::left_join(gene_exp_df, by = c("barcodes", "library_id"))
+```
+
+
+## Clustering
+
+Below we perform Louvain, Jaccard clustering, varying `k`. 
+The minimum `k` is 5 and the maximum `k` is 40 with a step size of 5. 
+
+```{r}
+# perform clustering across k = 5-40 with increments of 5
+all_cluster_results <- sce_list |>
+  purrr::map(cluster_sweep) |>
+  dplyr::bind_rows(.id = "library_id")
+```
+
+```{r}
+# get umap embeddings and combine into a data frame with cluster assignments
+umap_df <- sce_list |>
+  purrr::map(\(sce){
+    df <- sce |>
+      scuttle::makePerCellDF(use.dimred = "UMAP") |>
+      # replace UMAP.1 with UMAP1 and get rid of excess columns
+      dplyr::select(barcodes, UMAP1 = UMAP.1, UMAP2 = UMAP.2)
+  }) |>
+  dplyr::bind_rows(.id = "library_id") |>
+  dplyr::left_join(all_cluster_results, by = c("library_id", "barcodes"))
+```
+
+Below we visualize the cluster assignments using each parameter on a UMAP. 
+This isn't particularly useful because of clusters, but it can be helpful to see any values of K that may have obvious over or under clustering. 
+
+```{r, fig.height=15, fig.width=10}
+# look at clustering results across all params
```

Comment:
For this plot you're going to need to tweak the factors, since `k_5` is last because the computer thinks 5>40 😞

Reply:
Actually also, this plot is really hard to take in because it's so long. Why don't we just plot the two datasets separately so each has its own `facet_wrap`? We aren't comparing dataset UMAPs directly to each other anyways and a shared legend isn't particularly meaningful.